seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/SeaTunnelServer.java

```diff
@@ -238,7 +340,18 @@ public JobStatus getJobStatus(long jobId) {
             // TODO Get Job Status from JobHistoryStorage
             return JobStatus.FINISHED;
         }
-        return runningJobMaster.getJobStatus();
+        // This method is called by operation and in the runningJobMaster.getJobStatus() we will get data from IMap.
+        // It will cause an error "Waiting for response on this thread is illegal". To solve it we need put
+        // runningJobMaster.getJobStatus() in another thread.
+        CompletableFuture<JobStatus> future = CompletableFuture.supplyAsync(() -> {
+            return runningJobMaster.getJobStatus();
+        }, executorService);
```

Comment:
I think the problem is mainly caused by some limitations of the caller of the getJobStatus method, so solving this problem should be solved by the caller. instead of being handled internally by the method. Because calling in a normal thread should not encounter this problem.